lib/Service/Gateway/SMS/Provider/SpryngSMS.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Ruben de Wit <ruben@iamit.nl>
+ *
+ * Nextcloud - Two-factor Gateway
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\TwoFactorGateway\Service\Gateway\SMS\Provider;
+
+use Exception;
+use OCA\TwoFactorGateway\Exception\SmsTransmissionException;
+use OCP\Http\Client\IClient;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+
+class SpryngSMS implements IProvider {
+
+	const PROVIDER_ID = 'spryng';
+
+	/** @var IClient */
+	private $client;
+
+	/** @var SpryngSMSConfig */
+	private $config;
+
+	public function __construct(IClientService $clientService,
+								SpryngSMSConfig $config) {
+		$this->client = $clientService->newClient();
+		$this->config = $config;
+	}
+
+	/**
+	 * @param string $identifier
+	 * @param string $message
+	 *
+	 * @throws SmsTransmissionException
+	 */
+	public function send(string $identifier, string $message) {
+		$config = $this->getConfig();
```

Comment:
you might want to add a `/** @var SpryngSMSConfig $providerConfig */` line here as well, otherwise the `getApiToken` invocation below is problematic from a static point of view.

Reply:
I have added the comment you suggested